### ISSUE TITLE
[BUG] backupTarget / backupTargetCredentialSecret are never applied

### ISSUE THREAD

**Issue by EugenMayer** [longhorn/longhorn#3458]
Using this `values.yaml`

```yaml
defaultSettings:
  defaultReplicaCount: {replicas}
  backupTarget: "s3://longhorn@de/"
  backupTargetCredentialSecret: longhorn-minio-creds
  nodeDownPodDeletionPolicy: "delete-both-statefulset-and-deployment-pod"
  allowNodeDrainWithLastHealthyReplica: true
persistence:
  defaultClassReplicaCount: ${replicas}
```

It seems like most setting are applied, but backupTarget / backupTargetCredentialSecret are never applied. Tested this with a change on an existing cluster or an a new cluster. With a new cluster, both values are empty, with an existing, the values never change.

Considering how critical both values are, I would mark this one as critical.

Hint: I'am using 1.2.3 and i'am fairly confident this was working with 1.2.2 - any changes to configuration management (configMap replaced with CR introducing a bug?)

## ENV

- RKE2 latest
- longhorn 1.2.3 (helm based installation)

**Comment by EugenMayer** [comment]
Checked the deployed `BackupTarget` CR:

```
apiVersion: longhorn.io/v1beta1
kind: BackupTarget
metadata:
  creationTimestamp: '2021-12-23T09:14:38Z'
  generation: 1
  managedFields:
    - apiVersion: longhorn.io/v1beta1
      fieldsType: FieldsV1
      fieldsV1:
        f:spec: {}
      manager: longhorn-manager
      operation: Update
      time: '2021-12-23T09:14:38Z'
    - apiVersion: longhorn.io/v1beta1
      fieldsType: FieldsV1
      fieldsV1:
        f:status: {}
      manager: longhorn-manager
      operation: Update
      subresource: status
      time: '2021-12-23T09:15:36Z'
  name: default
  namespace: longhorn-system
  resourceVersion: '307456'
  uid: 3a07d17b-6d39-48a7-9870-532616f7565b
  selfLink: /apis/longhorn.io/v1beta1/namespaces/longhorn-system/backuptargets/default
spec:
  backupTargetURL: ''
  credentialSecret: ''
  pollInterval: 5m0s
  syncRequestedAt: null
```

So it is in fact not set at all

**Comment by jenting** [comment]
We have an issue worked on fixing this issue https://github.com/longhorn/longhorn/issues/2570.

**Comment by jenting** [comment]
> Hint: I'am using 1.2.3 and i'am fairly confident this was working with 1.2.2 - any changes to configuration management (configMap replaced with CR introducing a bug?)

Not, we haven't released this change on the Longhorn v1.2.3.

**Comment by jenting** [comment]
Let's use #2570 to track this issue, close this one.
